docs/BUILD.md

```diff
@@ -1,6 +1,56 @@
 # Building s2n-tls
 
-s2n-tls can be built as follows:
+To use s2n-tls, you must build the library from the source and then include it in your program.
+## Requirements
+
+s2n-tls supports and tests on **x86** and **arm** architectures.
+
+### System requirements
+
+* 20GB RAM available
+
+### Supported OS and Distributions: 
+
+| OS     | Distros |      |        |
+|--------|---------|------|--------|
+| Redhat | Fedora  | AL   |
+| debian | ubuntu  |
+| *BSD   | free    | open | darwin |
+
+s2n-tls does not support [Windows](https://github.com/aws/s2n-tls/issues/497).
```

Comment:
nit - I'm not sure the link belongs on "windows" here. Maybe link the whole sentence? Or maybe just put the issue link in by itself since github markdown formats it nicely:
```suggestion
s2n-tls does not support Windows: https://github.com/aws/s2n-tls/issues/497.
```